pkg/k8s/pipelineRun.go

```diff
@@ -152,7 +146,18 @@ func (r *pipelineRun) FinishState() (*api.StateItem, error) {
 	his := r.apiObj.Status.StateHistory
 	his = append(his, state)
 	r.apiObj.Status.StateHistory = his
-	return &state, r.updateStatus()
+	r.apiObj.Status.StateDetails = state
+	return &state
+}
+
+// FinishState set end time stamp of the current (defined) state and add it to the history
+// If no current state is defined a new state (A) with creation time of the PipelineRun as start time is created.
+// State is set to api.StateFinished
+// Returns the state details
+func (r *pipelineRun) FinishState() (*api.StateItem, error) {
```

Comment:
Why do we have `FinishState` at all? Why don't we use `UpdateState` (maybe with minor modifications)?